src/Rule/CreditCard.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Particle.
+ *
+ * @link      http://github.com/particle-php for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Particle (http://particle-php.com)
+ * @license   https://github.com/particle-php/validator/blob/master/LICENSE New BSD License
+ */
+namespace Particle\Validator\Rule;
+
+use byrokrat\checkdigit\Luhn;
+use Particle\Validator\Rule;
+
+/**
+ * This rule is for validating if a value is a valid credit card number.
+ *
+ * @package Particle\Validator\Rule
+ */
+class CreditCard extends Rule
+{
+    /**
+     * A constant that will be used when the value is not a valid credit card number.
+     */
+    const INVALID_FORMAT = 'CreditCard::INVALID_FORMAT';
+    const INVALID_CHECKSUM = 'CreditCard::INVALID_CHECKSUM';
+
+    /**
+     * The message templates which can be returned by this validator.
+     *
+     * @var array
+     */
+    protected $messageTemplates = [
+        self::INVALID_FORMAT => '{{ name }} must have a valid credit card number format',
+        self::INVALID_CHECKSUM => '{{ name }} must be a valid credit card number',
+    ];
+
+    /**
+     * Validates if the value is a valid credit card number.
+     *
+     * @param mixed $value
+     * @return bool
+     */
+    public function validate($value)
+    {
+        if (!$this->validateFormat($value)) {
+            return $this->error(self::INVALID_FORMAT);
+        } elseif (!$this->validateChecksum($value)) {
+            return $this->error(self::INVALID_CHECKSUM);
+        }
+
+        return true;
+    }
+
+    /**
+     * @param $value
+     *
+     * @return bool
+     */
+    private function validateChecksum($value)
+    {
+        $luhn = new Luhn();
+
+        return $luhn->isValid($value);
+    }
+
+    /**
+     * @param $value
+     *
+     * @return bool
+     */
+    private function validateFormat($value)
+    {
+        if (preg_match('/^4[0-9]{12}(?:[0-9]{3})?$/', $value) === 1) {
```

Comment:
Maybe you could move the regexes to a protected property of this class (`$validationRegexes`?), and loop over them like:

```
foreach ($this->validationRegexes as $regex) {
   if (preg_match($regex, $value)) {
      return true;
   }
}
return false;
```

Would probably be easier to add new ones if necessary ;)